src/workerd/io/worker-entrypoint.c++

```diff
@@ -332,6 +341,14 @@ kj::Promise<void> WorkerEntrypoint::request(kj::HttpMethod method,
       failOpenService = context.getSubrequestChannelNoChecks(
           IoContext::NEXT_CLIENT_CHANNEL, false, kj::mv(cfBlobJson));
     }
+
+    if (proxyTask == kj::none && !loggedExceptionEarlier) {
```

Comment:
Probably worth a comment to say what we're checking here: We only want to run this logic if the request was canceled from the client side, not if it completed successfully or failed with an exception.